packages/services/api/src/modules/schema/providers/artifact-storage-writer.ts

```diff
@@ -24,7 +25,7 @@ const artifactMeta = {
  * Write an Artifact to an S3 bucket.
  */
 export class ArtifactStorageWriter {
-  constructor(private s3Client: S3Client, private bucketName: string) {}
+  constructor(@Inject(S3_CONFIG) private s3: S3Config) {}
```

Comment:
@n1ru4l you can create a wrapping Provider (as class) if you want to avoid all the manually typing + Inject + using the branded token